### ISSUE TITLE
Undocumented: a loss graph may ignore its target, and PerElement works inside a [Module]

### ISSUE THREAD

**Issue by Eric-Louchez** [Shorokoo/Shorokoo#203]
`TrainingRig`'s loss slot is exactly `(predictions, targets) -> Scalar<float32>`, so the model must
output whatever the loss consumes. For a language model that means emitting the `[B·L, V]` logit
tensor as a graph *output* — 6.6 GB at batch 32 with a 50 K vocabulary — purely to hand it to a
cross-entropy module.

The way out is to compute the loss inside the model and make the rig's "loss" a pass-through:

```csharp
[Module] public partial class MeanOfPredictions {
    public static Scalar<float32> Inline(Tensor<float32> predictions, Tensor<float32> targets)
        => predictions.Reduce(ReduceKind.Mean, null, keepDims: false).Scalar();
}
```

That works — I verified the rig accepts a loss graph that **ignores its target input entirely**, and
`TrainStep` runs and reports a loss. But it is undocumented, so a user has no way to know it is
legal, and the adapter module is dead weight that exists only to satisfy the signature.

Two asks:

1. Document that a loss graph may ignore its `targets` input, and that `PerElement`/`Reduced` — which
   `nn-library.md` correctly says are not rig-bindable — work fine as ordinary graph ops inside a
   `[Module]` body. That combination is the normal shape for a language model and is currently
   something you have to discover.
2. Consider allowing a model graph that already returns a scalar loss to be used without an adapter.
